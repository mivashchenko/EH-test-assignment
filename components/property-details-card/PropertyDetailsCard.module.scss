
.root {
  display: flex;
  flex-direction: column;
  background: $white;
  border-radius: 16px;
  padding: 32px 24px 32px 24px;
  gap: 8px;
  border: 1px solid #D0D2D3;

  position: relative;
  top: -32px;
  left: 16px;
  right: 16px;
  width: calc(100% - 32px);

  box-shadow: 4px 8px 8px 0px #00000040;

}

.header {
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
}

.headerClosing {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  justify-content: flex-end;

  &Label {
    font-size: 12px;
    font-weight: $fontWeightRegular;
    color: $grey3;
    text-transform: uppercase;
  }

  &Value {
    font-size: 16px;
    font-weight: $fontWeightBold;
    color: $grey3;
    text-transform: uppercase;
    margin-top: 4px;
  }
}

.price {
  font-size: 30px;
  font-weight: $fontWeightBold;
  color: $grey3;
}

.address {
  font-size: 14px;
  font-weight: $fontWeightBold;
  line-height: 14px;
  text-align: left;
  color: $grey3;
}

.description {
  font-size: 12px;
  font-weight: $fontWeightLight;
  line-height: 12px;
  text-align: left;
  font-variant: small-caps;

  color: $grey3;
}

.iconLabelContainer {
  display: flex;
  justify-content: space-between;
  align-items: center;
  background: $white;
}

.buttonsContainer {
  display: flex;
  flex-direction: column;
  padding: 16px 0;
  gap: 8px;

  &_block {
    display: flex;
    justify-content: space-between;
    gap: 8px;
  }
}

.dataGridContainer {
  width: 100%;
  border: 1px solid $grey1;
  border-radius: 8px;
  background: rgba($primary, 0.1);
}

.dataGridRow {
  width: 100%;
  display: flex;
  border-bottom: 1px solid $grey1;

  &:last-child {
    border-bottom: none;
  }


  &Cell {
    flex: 1;
    display: flex;
    flex-direction: column;
    padding: 8px;
  }
}

.dataGridRowCellLabel {
  font-size: 12px;
  font-weight: $fontWeightRegular;
  color: $grey3;
  text-transform: uppercase;
}

.dataGridRowCellValue {
  font-size: 16px;
  font-weight: $fontWeightBold;
  color: $grey3;
  text-transform: uppercase;
  margin-top: 4px;
}


@media only screen and (max-width: $screenTablet) {
  .dataGridRow {
    flex-wrap: wrap;
  }

  .dataGridRowCell {
    flex: 1 1 100%;

    &:nth-child(3) {
      border-top: 1px solid $grey1;
    }

  }

  .dataGridRowCellLabel {
    text-align: center;
  }

  .dataGridRowCellValue {
    text-align: center;
  }


}

