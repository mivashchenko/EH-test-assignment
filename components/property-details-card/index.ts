export {
  PropertyDetailsCard,
  PropertyDetailsCardPrice,
  PropertyDetailsCardAddress,
  PropertyDetailsCardDescription,
  PropertyDetailsCardIconLabelContainer,
  PropertyDetailsCardButtonContainer,
} from './property-details-card'
