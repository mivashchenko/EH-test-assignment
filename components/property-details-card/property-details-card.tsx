import styles from './style.module.scss'
import { ReactNode } from 'react'
import { Divider } from '../ui/divider'

const PropertyDetailsCard = ({ children }: { children: ReactNode }) => {
  return <div className={styles.root}>{children}</div>
}

const PropertyDetailsCardPrice = ({ content }: { content: number }) => {
  const formattedPrice = new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: 'USD',
    maximumSignificantDigits: 3,
  }).format(content)

  return <h3 className={styles.price}>{formattedPrice}</h3>
}

const PropertyDetailsCardAddress = ({ content }: { content: string }) => {
  return <p className={styles.address}>{content}</p>
}

const PropertyDetailsCardDescription = ({ content }: { content: string }) => {
  return <p className={styles.description}>{content}</p>
}

const PropertyDetailsCardIconLabelContainer = ({
  children,
}: {
  children: ReactNode
}) => {
  return (
    <div>
      <Divider direction={'horizontal'} />
      <div className={styles.iconLabelContainer}>{children}</div>
      <Divider direction={'horizontal'} />
    </div>
  )
}

interface PropertyDetailsCardButtonContainerProps {
  topButtons: ReactNode
  bottomButtons: ReactNode
}

const PropertyDetailsCardButtonContainer = ({
  topButtons,
  bottomButtons,
}: PropertyDetailsCardButtonContainerProps) => {
  return (
    <div className={styles.buttonsContainer}>
      <button className={styles.buttonsContainer_block}>{topButtons}</button>
      <button className={styles.buttonsContainer_block}>{bottomButtons}</button>
    </div>
  )
}

export {
  PropertyDetailsCard,
  PropertyDetailsCardPrice,
  PropertyDetailsCardAddress,
  PropertyDetailsCardDescription,
  PropertyDetailsCardIconLabelContainer,
  PropertyDetailsCardButtonContainer,
}
