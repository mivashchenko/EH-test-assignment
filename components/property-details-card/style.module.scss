
.root {
  display: flex;
  flex-direction: column;
  background: $white;
  border-radius: 16px;
  padding: 32px 24px 32px 24px;
  gap: 8px;
  border: 1px solid #D0D2D3;

  position: relative;
  top: -32px;
  left: 16px;
  right: 16px;
  width: calc(100% - 32px);

  box-shadow: 4px 8px 8px 0px #00000040;

}

.price {
  font-size: 30px;
  font-weight: $fontWeightBold;
  color: $grey3;
}

.address {
  font-size: 14px;
  font-weight: $fontWeightBold;
  line-height: 14px;
  text-align: left;
  color: $grey3;
}

.description {
  font-size: 12px;
  font-weight: $fontWeightLight;
  line-height: 12px;
  text-align: left;
  font-variant: small-caps;

  color: $grey3;
}

.iconLabelContainer {
  display: flex;
  justify-content: space-between;
  align-items: center;
  background: $white;
}

.buttonsContainer {
  display: flex;
  flex-direction: column;

  &_block {
    display: flex;
    justify-content: space-between;
    gap: 8px;
  }
}
