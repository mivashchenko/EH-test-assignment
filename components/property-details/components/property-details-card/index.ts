export {
  PropertyDetailsCard,
  PropertyDetailsCardHeader,
  PropertyDetailsCardHeaderClosing,
  PropertyDetailsCardPrice,
  PropertyDetailsCardAddress,
  PropertyDetailsCardDescription,
  PropertyDetailsCardIconLabelContainer,
  PropertyDetailsCardButtonContainer,
  PropertyDetailsCardButtonContainerBlock,
  PropertyDetailsCardDataGridContainer,
  PropertyDetailsCardDataGridRow,
  PropertyDetailsCardDataGridRowCell,
  PropertyDetailsCardDataGridRowCellLabel,
  PropertyDetailsCardDataGridRowCellValue,
} from './property-details-card'
