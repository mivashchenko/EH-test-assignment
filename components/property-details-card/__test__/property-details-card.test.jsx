import { render, screen } from '@testing-library/react'
import { PropertyDetailsCard } from '@/components/property-details-card/property-details-card'
import { PropertyDetailsCardHeader } from '@/components/property-details-card/property-details-card'
import { PropertyDetailsCardHeaderClosing } from '@/components/property-details-card/property-details-card'
import { PropertyDetailsCardPrice } from '@/components/property-details-card/property-details-card'
import { PropertyDetailsCardAddress } from '@/components/property-details-card/property-details-card'
import { PropertyDetailsCardDescription } from '@/components/property-details-card/property-details-card'
import { PropertyDetailsCardIconLabelContainer } from '@/components/property-details-card/property-details-card'
import { PropertyDetailsCardButtonContainer } from '@/components/property-details-card/property-details-card'
import { PropertyDetailsCardButtonContainerBlock } from '@/components/property-details-card/property-details-card'
import { PropertyDetailsCardDataGridContainer } from '@/components/property-details-card/property-details-card'
import { PropertyDetailsCardDataGridRowCell } from '@/components/property-details-card/property-details-card'
import { PropertyDetailsCardDataGridRowCellLabel } from '@/components/property-details-card/property-details-card'
import { PropertyDetailsCardDataGridRowCellValue } from '@/components/property-details-card/property-details-card'

it('should render a div with the provided children', () => {
  const { container } = render(
    <PropertyDetailsCard>
      <span>Test Child</span>
    </PropertyDetailsCard>
  )
  expect(container.firstChild).toHaveClass('root')
  expect(container.firstChild).toContainHTML('<span>Test Child</span>')
})

it('should render correctly even without any children', () => {
  const { container } = render(<PropertyDetailsCard />)
  expect(container.firstChild).toHaveClass('root')
  expect(container.firstChild).toBeEmptyDOMElement()
})

it('should render without crashing with valid ReactNode children', () => {
  const { container } = render(
    <PropertyDetailsCardHeader>
      <div>Test Child</div>
    </PropertyDetailsCardHeader>
  )
  expect(container).toBeTruthy()
})

it('should render correctly with no children', () => {
  const { container } = render(<PropertyDetailsCardHeader />)
  expect(container.firstChild).toBeEmptyDOMElement()
})

it('should render label and value correctly when both are provided', () => {
  const { getByText } = render(
    <PropertyDetailsCardHeaderClosing
      label='Closing Date'
      value='05 October 2011 14:48 UTC'
    />
  )
  expect(getByText('Closing Date')).toBeInTheDocument()
  expect(getByText('October 5, 2011')).toBeInTheDocument()
})

it('should render nothing when both label and value are empty', () => {
  const { container } = render(
    <PropertyDetailsCardHeaderClosing label='' value='' />
  )
  expect(container.firstChild).toBeNull()
})

it('should format positive numbers as USD currency', () => {
  const { container } = render(<PropertyDetailsCardPrice content={1500} />)
  expect(container.textContent).toBe('$1,500')
})

it('should handle extremely large numbers correctly', () => {
  const { container } = render(
    <PropertyDetailsCardPrice content={123456789012345} />
  )
  expect(container.textContent).toBe('$123,000,000,000,000')
})

it('should display the address content correctly when provided with a typical address string', () => {
  const { getByText } = render(
    <PropertyDetailsCardAddress content='123 Main St, Anytown, USA' />
  )
  expect(getByText('123 Main St, Anytown, USA')).toBeInTheDocument()
})

it('should display nothing when provided with an empty string', () => {
  const { container } = render(<PropertyDetailsCardAddress content='' />)
  expect(container.firstChild).toBeEmptyDOMElement()
})

it('should render a paragraph element with the provided content', () => {
  const { container } = render(
    <PropertyDetailsCardDescription content='Test content' />
  )
  const paragraph = container.querySelector('p')
  expect(paragraph).toBeInTheDocument()
  expect(paragraph).toHaveTextContent('Test content')
})

it('should handle empty string content without errors', () => {
  const { container } = render(<PropertyDetailsCardDescription content='' />)
  const paragraph = container.querySelector('p')
  expect(paragraph).toBeInTheDocument()
  expect(paragraph).toHaveTextContent('')
})

it('should display only dividers when rendered with no children', () => {
  const { container } = render(<PropertyDetailsCardIconLabelContainer />)
  expect(container.firstChild).toMatchSnapshot()
})

it('should handle null children without crashing', () => {
  const { container } = render(
    <PropertyDetailsCardIconLabelContainer>
      {null}
    </PropertyDetailsCardIconLabelContainer>
  )
  expect(container.firstChild).toMatchSnapshot()
})

it('should render without crashing', () => {
  const { container } = render(
    <PropertyDetailsCardButtonContainer>
      <div>Test</div>
    </PropertyDetailsCardButtonContainer>
  )
  expect(container).toBeInTheDocument()
})

it('should handle null children without errors', () => {
  const { container } = render(
    <PropertyDetailsCardButtonContainer>
      {null}
    </PropertyDetailsCardButtonContainer>
  )
  expect(container).toBeInTheDocument()
})

it('should render a div with the specified children', () => {
  const { container } = render(
    <PropertyDetailsCardButtonContainerBlock>
      <span>Test Child</span>
    </PropertyDetailsCardButtonContainerBlock>
  )
  expect(container.firstChild).toHaveClass('buttonsContainer_block')
  expect(container.firstChild).toContainHTML('<span>Test Child</span>')
})

it('should render correctly without any children', () => {
  const { container } = render(<PropertyDetailsCardButtonContainerBlock />)
  expect(container.firstChild).toHaveClass('buttonsContainer_block')
  expect(container.firstChild).toBeEmptyDOMElement()
})

it('should render a div with the specified children', () => {
  const { container } = render(
    <PropertyDetailsCardDataGridContainer>
      <span>Test Child</span>
    </PropertyDetailsCardDataGridContainer>
  )
  expect(container.firstChild).toHaveClass('dataGridContainer')
  expect(container.firstChild).toContainHTML('<span>Test Child</span>')
})

it('should render correctly with no children', () => {
  const { container } = render(<PropertyDetailsCardDataGridContainer />)
  expect(container.firstChild).toHaveClass('dataGridContainer')
  expect(container.firstChild).toBeEmptyDOMElement()
})

it('should render a div with the specified children', () => {
  const { container } = render(
    <PropertyDetailsCardDataGridRowCell>
      <span>Test Child</span>
    </PropertyDetailsCardDataGridRowCell>
  )
  expect(container.firstChild).toHaveClass('dataGridRowCell')
  expect(container.firstChild).toContainHTML('<span>Test Child</span>')
})

it('should render correctly without any children', () => {
  const { container } = render(<PropertyDetailsCardDataGridRowCell />)
  expect(container.firstChild).toHaveClass('dataGridRowCell')
  expect(container.firstChild).toBeEmptyDOMElement()
})

it('should render without crashing', () => {
  const { unmount } = render(<PropertyDetailsCardDataGridRowCellLabel />)
  unmount()
})

it('should render correctly with no children', () => {
  const { container } = render(<PropertyDetailsCardDataGridRowCellLabel />)
  expect(container.firstChild).toBeEmptyDOMElement()
})

it('should render a div with the specified children', () => {
  const { container } = render(
    <PropertyDetailsCardDataGridRowCellValue>
      <span>Test Child</span>
    </PropertyDetailsCardDataGridRowCellValue>
  )
  expect(container.querySelector('div')).toContainHTML(
    '<span>Test Child</span>'
  )
})

it('should render correctly with no children passed', () => {
  const { container } = render(<PropertyDetailsCardDataGridRowCellValue />)
  expect(container.querySelector('div')).toBeEmptyDOMElement()
})
