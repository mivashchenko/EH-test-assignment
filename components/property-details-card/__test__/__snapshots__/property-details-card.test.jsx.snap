// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display only dividers when rendered with no children 1`] = `
<div>
  <hr
    class="root horizontal"
  />
  <div
    class="iconLabelContainer"
  />
  <hr
    class="root horizontal"
  />
</div>
`;

exports[`should handle null children without crashing 1`] = `
<div>
  <hr
    class="root horizontal"
  />
  <div
    class="iconLabelContainer"
  />
  <hr
    class="root horizontal"
  />
</div>
`;
